<a href="https://colab.research.google.com/github/Anshuman-Tiwari-2002/DS-project/blob/main/Day2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Features of Dataset

In [4]:
import pandas as pd

In [5]:
df = pd.read_csv('https://talentcocomedia.s3.amazonaws.com/ml-assets/heart-disease.csv')

In [6]:
df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [7]:
df.shape #rows, colums

(303, 14)


1.age :- current age of the patient

2.sex :- Gender of the patient (1 refers to Male and 0 refers to Fema3le)

3.cp :- Refers to the chest pain type... a} 0 : Typical Angina :: is defined as substernal chest pain precipitated by physical exertion or emotional stress and relieved with rest or nitroglycerin. b} 1 : Atypical Angina :: Angina chest pain is a pressure or squeezing like sensation that is usually caused when your heart muscle doesn't get an adequate supply of oxygenated blood. c} 2 : Non-Anginal Pain :: defined as recurrent chest pain that is indistinguishable from ischemic heart pain after a reasonable workup has excluded a cardiac cause. d} 3 : Asymptomatic Pain :: Silent (asymptomatic) myocardial ischemia (SMI) is defined as a transient alteration in myocardial perfusion in the absence of chest pain or the usual anginal equivalents.

4.trestbps :- The person's resting blood pressure (mm Hg on admission to the hospital)

5.chol :- serum cholestrol in mg/DL

6.fbs :- fasting blood sugar > 120 mg/dl (1 = true; 0 = false)

7.restecg :- resting electrocardiographic results (0 = normal; 1 = having ST-T; 2 = hypertrophy)

8.thalach :- maximum heart rate achieved.

9.exang :- exercise-include angina (True/False)

10.oldpeak :- ST depression induced by exercise relative to rest

11.slope :- The slope of the peak exercise ST segm:ent

12.ca :- number of major vessels (0-3) colored by fluoroscopy

13.thal :- normal, fixed defect, reversible defect (thalasemia indicator)

14.target :- the predicted attribute



